pkg/controller/test-images-distributor/test_images_distributor.go

```diff
@@ -543,3 +548,12 @@ func upsertObject(ctx context.Context, c ctrlruntimeclient.Client, obj crcontrol
 	}
 	return err
 }
+
+func isImportAllowed(pullString string) bool {
+	return strings.HasPrefix(pullString, "registry.svc.ci.openshift.org") ||
+		strings.HasPrefix(pullString, "docker-registry.default.svc:5000") ||
+		strings.HasPrefix(pullString, "registry.access.redhat.com") ||
+		strings.HasPrefix(pullString, "quay.io") ||
+		strings.HasPrefix(pullString, "gcr.io") ||
+		strings.HasPrefix(pullString, "docker.io")
+}
```

Comment:
Since this is a pretty intense loop, it's worth making this into a set:

<details>

```
$ cat > a_test.go <<'EOF'
package main

import (
	"strings"
	"testing"
)

func isImportAllowedPrefix(pullString string) bool {
	return strings.HasPrefix(pullString, "registry.svc.ci.openshift.org") ||
		strings.HasPrefix(pullString, "docker-registry.default.svc:5000") ||
		strings.HasPrefix(pullString, "registry.access.redhat.com") ||
		strings.HasPrefix(pullString, "quay.io") ||
		strings.HasPrefix(pullString, "gcr.io") ||
		strings.HasPrefix(pullString, "docker.io")
}

var m = map[string]struct{}{
	"registry.svc.ci.openshift.org": {},
	"docker-registry.default.svc:5000": {},
	"registry.access.redhat.com": {},
	"quay.io": {},
	"gcr.io": {},
	"docker.io": {},
}

func isImportAllowedSet(pullString string) bool {
	i := strings.Index(pullString, "/")
	var ok bool
	if i == -1 { // whatever
		_, ok = m[pullString]
	} else {
		_, ok = m[pullString[:i]]
	}
	return ok
}

var positive = "quay.io/ci-op-hbtwhrrm/pipeline@sha256:328d0a90295ef5f5932807bcab8f230007afeb1572d1d7878ab8bdae671dfa8b"
var negative = "404/ci-op-hbtwhrrm/pipeline@sha256:328d0a90295ef5f5932807bcab8f230007afeb1572d1d7878ab8bdae671dfa8b"

func BenchmarkPrefixPositive(b *testing.B) {
	for i := 0; i < b.N; i++ { isImportAllowedPrefix(positive) }
}

func BenchmarkPrefixNegative(b *testing.B) {
	for i := 0; i < b.N; i++ { isImportAllowedPrefix(negative) }
}

func BenchmarkSetPositive(b *testing.B) {
	for i := 0; i < b.N; i++ { isImportAllowedSet(positive) }
}

func BenchmarkSetNegative(b *testing.B) {
	for i := 0; i < b.N; i++ { isImportAllowedSet(negative) }
}
EOF
$ go test --bench .
goos: linux
goarch: amd64
BenchmarkPrefixPositive-8       71225979                17.2 ns/op
BenchmarkPrefixNegative-8       42517066                28.3 ns/op
BenchmarkSetPositive-8          75177789                16.1 ns/op
BenchmarkSetNegative-8          80234144                15.0 ns/op
```

</details>